src/geometry/vector2d.ts

```diff
@@ -0,0 +1,157 @@
+import { Integer, int } from "utils";
+
+export enum Symmetry {
+  ORIGIN = 0,
+  X_AXIS = 1,
+  Y_AXIS = 2
+}
+
+
+export class Vector2d {
+  protected readonly x: number;
+  protected readonly y: number;
+
+  protected createVector(x: number, y: number): Vector2d {
+    return new Vector2d(x, y);
+  }
+
+  public constructor(x: number, y: number) {
+    this.x = x;
+    this.y = y;
+  }
+
+  public getX() {
+    return this.x;
+  }
+
+  public getY() {
+    return this.y;
+  }
+
+  public toString() {
+    return this.x + ", " + this.y;
+  }
+
+  public equals(other: unknown): boolean {
+    if (!(other instanceof Vector2d)) {
+      return false;
+    }
+    return this.x == other.x && this.y == other.y;
+  }
+
+  public add(other: Vector2d): Vector2d {
+    return this.createVector(this.x + other.x, this.y + other.y);
+  }
+
+  public sub(other: Vector2d): Vector2d {
+    return this.createVector(this.x - other.x, this.y - other.y);
+  }
+
+  public mul(other: Vector2d | number): Vector2d {
+    if (other instanceof Vector2d) {
+      return this.createVector(this.x * other.x, this.y * other.y);
+    }
+    return this.createVector(this.x * other, this.y * other);
+  }
+
+  public div(other: Vector2d | number): Vector2d {
+    if (other instanceof Vector2d) {
+      if(other.x == 0 || other.y == 0){
+        throw new Error("At least one coordinate of divider vector is equal to 0.");
+      }
```

Comment:
Should split the condition into separate dimensions for the clarity of exception. Also, add space after `if` keyword and closing bracket.